langchain/src/document_loaders/github.ts

```diff
@@ -75,48 +101,52 @@ export class GithubRepoLoader extends BaseDocumentLoader {
               new Document({ pageContent: fileContent, metadata })
             );
           } catch (e) {
-            this.handleError(`Failed to fetch file content: ${file.path}`);
+            this.handleError(
+              `Failed to fetch file content: ${file.path}, ${e}`
+            );
           }
         }
       }
     } catch (error) {
-      this.handleError(`Failed to process directory: ${path}`);
+      this.handleError(`Failed to process directory: ${path}, ${error}`);
     }
   }
 
-  private async fetchRepoFiles(path: string): Promise<any[]> {
+  private async fetchRepoFiles(path: string): Promise<GithubFile[]> {
     const url = `https://api.github.com/repos/${this.owner}/${this.repo}/contents/${path}?ref=${this.branch}`;
     const response = await fetch(url);
     const data = await response.json();
+    if (!response.ok) {
+      throw new Error(
+        `Unable to fetch repository files: ${response.status} ${JSON.stringify(
+          data
+        )}`
+      );
+    }
 
     if (!Array.isArray(data)) {
       throw new Error("Unable to fetch repository files.");
     }
 
-    return data;
+    return data as GithubFile[];
   }
 
-  private async fetchFileContent(file: any): Promise<string> {
-    // Check if the file has a MIME type that starts with "image/" or "application/pdf"
+  private async fetchFileContent(file: GithubFile): Promise<string> {
+    const fileExtension = file.name.split(".").pop()?.toLowerCase();
     if (
-      file.mime &&
-      (file.mime.startsWith("image/") || file.mime === "application/pdf")
+      fileExtension === "jpg" ||
```

Comment:
Hmm I don't like this, if someone has some other kind of file we didn't think of (eg. docx) then we'll just let it go through?

Reply:
We can eg use this instead https://github.com/sindresorhus/is-binary-path or the opposite https://github.com/sindresorhus/is-text-path

Reply:
Yeah you're right, we also need to exclude lock files